src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -204,9 +205,24 @@ private void AddCompilationRoots()
                 AddCompilationRootsForRuntimeExports(module);
            }
 
+            AddCompilationRootsForBootstrap();
+
             AddCompilationRootsForRuntimeExports((EcmaModule)_typeSystemContext.SystemModule);
         }
 
+        private void AddCompilationRootsForBootstrap()
```

Comment:
This should be moved/merged into AddCompilationRootsForMainMethod(). You do not need to root the Main method anymore - it will be rooted indirectly by the bootstrapper.